instrumentation/rocketmq/rocketmq-client/rocketmq-client-5.0/testing/src/main/java/io/opentelemetry/instrumentation/rocketmqclient/v5_0/AbstractRocketMqClientTest.java

```diff
@@ -41,135 +49,312 @@
-                                .hasStatus(StatusData.unset())
-                                .hasAttributesSatisfyingExactly(
-                                    equalTo(MESSAGING_ROCKETMQ_CLIENT_GROUP, consumerGroup),
-                                    equalTo(MESSAGING_SYSTEM, "rocketmq"),
-                                    equalTo(
-                                        MESSAGING_DESTINATION_KIND,
-                                        SemanticAttributes.MessagingDestinationKindValues.TOPIC),
-                                    equalTo(MESSAGING_DESTINATION, topic),
-                                    equalTo(MESSAGING_OPERATION, "receive")),
-                        span ->
-                            span.hasKind(SpanKind.CONSUMER)
-                                .hasName(topic + " process")
-                                .hasStatus(StatusData.unset())
-                                // Link to send span.
-                                .hasLinks(LinkData.create(sendSpanData.get().getSpanContext()))
-                                // As the child of receive span.
-                                .hasParent(trace.getSpan(0))
-                                .hasAttributesSatisfyingExactly(
-                                    equalTo(MESSAGING_ROCKETMQ_CLIENT_GROUP, consumerGroup),
-                                    equalTo(MESSAGING_ROCKETMQ_MESSAGE_TAG, tag),
-                                    equalTo(MESSAGING_ROCKETMQ_MESSAGE_KEYS, Arrays.asList(keys)),
-                                    equalTo(
-                                        MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES, (long) body.length),
-                                    equalTo(MESSAGING_SYSTEM, "rocketmq"),
-                                    equalTo(
-                                        MESSAGING_MESSAGE_ID,
-                                        sendReceipt.getMessageId().toString()),
-                                    equalTo(
-                                        MESSAGING_DESTINATION_KIND,
-                                        SemanticAttributes.MessagingDestinationKindValues.TOPIC),
-                                    equalTo(MESSAGING_DESTINATION, topic),
-                                    equalTo(MESSAGING_OPERATION, "process")),
-                        span ->
-                            span.hasName("child")
-                                .hasKind(SpanKind.INTERNAL)
-                                .hasParent(trace.getSpan(1))));
-      }
+            .build();
+    producer =
+        provider
+            .newProducerBuilder()
+            .setClientConfiguration(clientConfiguration)
+            .setTopics(topic)
+            .build();
+  }
+
+  @AfterAll
+  void tearDown() throws IOException {
+    if (producer != null) {
+      producer.close();
+    }
+    if (consumer != null) {
+      // Not calling consumer.close(); because it takes a lot of time to complete
```

Comment:
the built-in threadpools don't shutdown if `consumer#close` is not called, or could we submit an async task to close consumer here?